src/util/chainsUtils.ts

```diff
@@ -24,14 +24,14 @@ interface Input {
 
 export function getSourceTargetRole({ useSourceTarget, sourceChain }: Input) {
   const {
-    sourceChainDetails: { sourceChain: currentSourceChain }
+    sourceChainDetails: { chain: currentSourceChain }
     // eslint-disable-next-line react-hooks/rules-of-hooks
   } = useSourceTarget();
 
   const sourceChainsMatch = sourceChain === currentSourceChain;
 
-  const sourceRole = sourceChainsMatch ? ChainDetails.SOURCE : ChainDetails.TARGET;
-  const targetRole = sourceChainsMatch ? ChainDetails.TARGET : ChainDetails.SOURCE;
+  const sourceRole = sourceChainsMatch ? ChainSubscriptions.SOURCE : ChainSubscriptions.TARGET;
+  const targetRole = sourceChainsMatch ? ChainSubscriptions.TARGET : ChainSubscriptions.SOURCE;
```

Comment:
That's a super confusing change. We have `getSourceTargetRole`, but instead we return `ChainSubscriptions`? So how to get `ChainDetails` now?
Why is fucntion called `sourceTargetRole` returning anything that deals with `subscriptions` anyway?
I'm surprised this doesn't break any existing code and the compiler is not able to catch this.

Reply:
Honestly i don't see how this would break if i took care of all places where this is used and updated accordingly: For example:  https://github.com/paritytech/parity-bridges-ui/pull/179/files#diff-8762bd5279e5304ba0047479b3927e524649384b2270f3023437979a5df37ed1R52

Now i understand your point about the change might sound confusing but in the end is returning the chain role  related to the source/target subscriptions. In case you find a better naming, please let me know and i will update them since i'm not convinced either.

Anyway, i think this util fly away after this: https://github.com/paritytech/parity-bridges-ui/issues/159

Reply:
> i took care of all places where this is used and updated according

Yes, then the name should signify it is dealing with subscriptions and `sourceTargetRole`. Like I honestly find it really hard to understand the reasoning behind returning roles and then extracting the correct connection/subscription based on the role.

Why can't we have a hook that returns the connection/subscriptions directly?

I agree that this is a bit OT, let's just change the name and refactor this during re-architecturing.

Reply:
ok, any recommendations for renaming?

Reply:
I'm not sold on the entire "role" thing - like this function literally returns a field name from subscriptions object. Then maybe let's call it `getChainSubscriptionsKey`?

Reply:
deal.